common/primitives/src/parquet/temporal.rs

```diff
@@ -0,0 +1,82 @@
+use codec::{Decode, Encode, MaxEncodedLen};
+use scale_info::TypeInfo;
+use serde::{Deserialize, Serialize};
+use sp_std::prelude::*;
+
+/// Parquet temporal types: https://github.com/apache/parquet-format/blob/master/LogicalTypes.md
+#[derive(
+	Clone, Encode, Decode, PartialEq, Debug, TypeInfo, Eq, MaxEncodedLen, Serialize, Deserialize,
+)]
+pub enum ParquetTemporalType {
+	/// Parquet dates
+	Date,
+	/// Parquet intervals
+	Interval,
+	/// Time
+	Time(ParquetTime),
+	/// Timestamps
+	Timestamp(ParquetTimestamp),
+}
+
+impl Default for ParquetTemporalType {
+	fn default() -> Self {
+		Self::Timestamp(ParquetTimestamp::default())
+	}
+}
+
+/// Parquet time: https://github.com/apache/parquet-format/blob/master/LogicalTypes.md
+#[derive(
+	Default,
+	Clone,
+	Encode,
+	Decode,
+	PartialEq,
+	Debug,
+	TypeInfo,
+	Eq,
+	MaxEncodedLen,
+	Serialize,
+	Deserialize,
+)]
+pub struct ParquetTime {
+	is_adjusted_to_utc: bool,
+	unit: ParquetTimeUnit,
+}
+
+/// Parquet timestamps: https://github.com/apache/parquet-format/blob/master/LogicalTypes.md
+#[derive(
+	Default,
+	Clone,
+	Encode,
+	Decode,
+	PartialEq,
+	Debug,
+	TypeInfo,
+	Eq,
+	MaxEncodedLen,
+	Serialize,
+	Deserialize,
+)]
+pub struct ParquetTimestamp {
+	is_adjusted_to_utc: bool,
+	unit: ParquetTimeUnit,
+}
+
+/// Units of time
+#[derive(
+	Clone, Encode, Decode, PartialEq, Debug, TypeInfo, Eq, MaxEncodedLen, Serialize, Deserialize,
+)]
+enum ParquetTimeUnit {
+	/// milliseconds
+	MILLIS,
+	/// microseconds
+	MICROS,
+	/// nanoseconds
+	NANOS,
+}
```

Comment:
could also include quick details like millis is int32 and the other two are int64
I know this comes from parquet type documentation, but could be worthwhile to include some details?

Reply:
I don't know if that particular type info is super useful for this enum (the person using it does not necessarily need to know the integer type for their schema)... but you're right maybe there is more information I can add.